openai-core/src/commonMain/kotlin/com.aallam.openai.api/assistant/AssistantTruncationStrategy.kt

```diff
@@ -0,0 +1,20 @@
+package com.aallam.openai.api.assistant
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+public data class AssistantTruncationStrategy(
+    /**
+     * The truncation strategy to use for the thread.
+     * The default is auto.
+     * If set to `last_messages`, the thread will be truncated to the n most recent messages in the thread.
+     * When set to `auto`, messages in the middle of the thread will be dropped to fit the context length of the model, `max_prompt_tokens`.
+     */
+    @SerialName("type") val type: String,
+
+    /**
+     * The number of most recent messages from the thread when constructing the context for the run.
+     */
+    @SerialName("last_messages") val lastMessages: Int? = null
+)
```

Comment:
Maybe we can improve a little bit the API here, with something like this:

```suggestion
    @SerialName("type") val type: TruncationStrategyType,

    /**
     * The number of most recent messages from the thread when constructing the context for the run.
     */
    @SerialName("last_messages") val lastMessages: Int? = null
)

/**
 * The truncation strategy to use for the thread.
 */
@JvmInline
@Serializable
public value class TruncationStrategyType(public val value: String) {
    public companion object {
        public val TruncationStrategyType: Auto = TruncationStrategyType("auto")
        public val TruncationStrategyType LastMessages = TruncationStrategyType("last_messages")
    }
}
```